geopandas/geodataframe.py

```diff
@@ -117,11 +117,20 @@ def __init__(self, *args, geometry=None, crs=None, **kwargs):
         # but within a try/except because currently non-geometries are
         # allowed in that case
         # TODO do we want to raise / return normal DataFrame in this case?
-        if geometry is None and "geometry" in self.columns:
+
+        # by default, if geometry is None, look for a geo col named geometry
+        assumed_geo_col_name = "geometry"
+
+        # if gdf passed in and geo_col is set, we use that for geometry,
+        # # instead of looking for a column named "geometry"
+        if geometry is None and isinstance(data, GeoDataFrame):
+            assumed_geo_col_name = data._geometry_column_name
```

Comment:
Could we directly `return` in this case? (instead of going through the whole set_geometry, which is not really needed if you already had a GeoDataFrame)

Reply:
> Could we directly `return` in this case? (instead of going through the whole set_geometry, which is not really needed if you already had a GeoDataFrame)

Yep, good catch.